update react-highlight-words package version to 0.18.0

Thank you for your attention to this issue. But there is another problem through installation.

Log is below:
```
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-highlight-words@0.12.0
npm WARN Found: react@17.0.2
npm WARN node_modules/react
npm WARN   react@"^17.0.2" from the root project
npm WARN   27 more (@devexpress/dx-react-chart, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react@"^0.14.0 || ^15.0.0 || ^16.0.0-0" from react-highlight-words@0.12.0
npm WARN node_modules/react-highlight-words
npm WARN   react-highlight-words@"^0.12.0" from react-dadata-box@1.3.7
npm WARN   node_modules/react-dadata-box
npm WARN 
npm WARN Conflicting peer dependency: react@16.14.0
npm WARN node_modules/react
npm WARN   peer react@"^0.14.0 || ^15.0.0 || ^16.0.0-0" from react-highlight-words@0.12.0
npm WARN   node_modules/react-highlight-words
npm WARN     react-highlight-words@"^0.12.0" from react-dadata-box@1.3.7
npm WARN     node_modules/react-dadata-box
```

You should increase your dependency package react-highlight-words to version 0.18, which is compatible with React 17 and 18. Or you can replace `^0.12.0` by `>=0.12`. You can check this page for version control of npm package: https://semver.npmjs.com/

_Originally posted by @vommuan in https://github.com/orbita-center/react-dadata-box/issues/68#issuecomment-1231128129_

--- a/README.md
+++ b/README.md
@@ -259,7 +259,8 @@ exported bulit-in types accordingly to type parameter
 
 ___
 #### forceOpenList ![](https://img.shields.io/badge/optional-green)
-this property force the suggestions list will be permanently open (usually needed for debug)
+this property force the already opened suggestions list will be permanently open, prevents close list by Blur event or user select value Action (it also is way may to debug suggestions)
+Attention! if query is empty and not declared any customActions - list is truthy empty of elements (empty list couldn't be open even if forceOpenList === true)
 ```typescript
 forceOpenList?: boolean;
 ```

--- a/README.md
+++ b/README.md
@@ -152,7 +152,15 @@ import { SpecificQueryModeResponse, FetchType } from 'react-dadata-box';
 type AbstractResponseType = CountryResponseType | PartyResponseType | BankResponseType | EmailResponseType | FioResponseType | FmsUnitResponseType
 
 // type of 'suggestion' wiil be infered automaticaliy from fetch type and it will be SpecificQueryModeResponse<T>[]
-onChange ? : (suggestion: SpecificQueryModeResponse<FetchType>) => void;
+onChange?: (suggestion: SpecificQueryModeResponse<FetchType>) => void;
+```
+___
+#### onBlur ![](https://img.shields.io/badge/optional-green)
+handler called on blur event of react-dadata-box input (also at closing list after user selection or outside click)
+this handler extended by second argument that contains current query text
+
+```typescript
+onBlur?: (event: SyntheticEvent<HTMLInputElement, FocusEvent>, query: string) => void;
 ```
 ___
 #### onIdleOut ![](https://img.shields.io/badge/optional-green)

--- a/README.ru.md
+++ b/README.ru.md
@@ -262,7 +262,8 @@ import { SpecificQueryModeResponse } from 'react-dadata-box';
  | 'fms_unit'  | FmsUnitResponseType |
 ___
 #### forceOpenList ![](https://img.shields.io/badge/optional-green)
-свойство определяющее принудительное раскрытие списка подсказок (преимущественно необходимо для удобства отладки/разработки н/п customActions)
+свойство определяющее поведение при котором раскрытый список саджестов не закрывается по событиям Blur и по действию выбора значения из списка (может использоваться для удобства отладки/разработки н/п customActions)
+Внимание! если query пустое и не определено ни одного customActions - список истино пуст (истино пустой список не раскрывается даже при forceOpenList === true)
 ```typescript
 forceOpenList?: boolean;
 ```

--- a/README.ru.md
+++ b/README.ru.md
@@ -158,6 +158,14 @@ type AbstractResponseType = CountryResponseType | PartyResponseType | BankRespon
 onChange?: (suggestion: SpecificQueryModeResponse<FetchType>) => void;
 ```
 ___
+#### onBlur ![](https://img.shields.io/badge/optional-green)
+обработчик события вызываемого при потере фокуса уомпонентом react-dadata-box (в т.ч. скрытие списка после пользовательского выбора и/или клика за пределы раскрытого списка)
+стандартный обработчик расширен вторым аргументом в который передается текущее занчение query (строки поиска, занчения в поле ввода)
+
+```typescript
+onBlur?: (event: SyntheticEvent<HTMLInputElement, FocusEvent>, query: string) => void;
+```
+___
 #### onIdleOut ![](https://img.shields.io/badge/optional-green)
 обработчик события вызываемого в случаях когда по текущему запросу пользователя сервис не возвращает подсказок, принимает в качестве аргумента текущую строку запроса.
 ```typescript

--- a/index.d.ts
+++ b/index.d.ts
@@ -380,6 +380,8 @@ interface BasePayload {
   count?: number;
 }
 
+
+
 /**
  * @typedef { BasePayload } BasePayload
  * @property  { string } className - [required] component CSS class name (default: native generated style with 'react-dadata__' prefix)
@@ -404,32 +406,40 @@ interface BaseInputProps<T = HTMLInputElement> {
 }
 
 /**
- * @typedef { SpecificModeByTypeProps } SpecificModeByTypeProps
- * @property  { boolean } autocomplete - [optional] property translated to native input tag;
- * @property  { customInput } customInput - [optional] function that fires with one argument { BaseInputProps }
- * @property  { customActions } customActions - [optional] adding custom action to base suggestions dropdown list
- * @property  { boolean | object } customEndpoint - [optional] optional uri to fetch suggestion's (to proxy scenario or local hosted DaData service), may be string: full or relative uri or object with 'host' and/or 'api' property
- * @property  { boolean } city - [optional] optional to city-mode
- * @property  { string } className - [optional] additional classname
- * @property  { number } count - [optional] single query limit (default: 10)
- * @property  { onChange } onChange - [optional] - onChange handler
- * @property  { onIdleOut } onIdleOut - [optional] - onIdleOut handler, fires with one argument - current query, when by this query has not returned suggestions
- * @property  { object | function } payloadModifier - [optional] - object to patch payload or function returned payload object to send, that has auto generated payload object as argument
- * @property  { debounce } onChange - [optional] - debounce to onChange handler (default: 350 ms).
- * @property  { string } placeholder - [optional] - placeholder
- * @property  { string } query - [optional] - query for search
- * @property  { React.CSSProperties } style - [optional] - custom styling
- * @property  { string } token - [required] - API authorization token
- * @property  { FetchType } type - [optional] specifics fetching by data type groups
- * @property  { boolean } allowClear - [optional] show/hide clear fieldd control
- * @property  { boolean } showNote - [optional] show/hide note at suggestions list
- * @property  { boolean } silentQuery - [optional] initial query that not showed on input but determine  suggestion list that showed at first
- * @property  { function } silentInit - [optional] function that may be used to autoselect from preventive fetched (by placed query or silentQuery), it called with list of fetched suggestions, and if it will return index, appropriate suggestion will be selected (all handlers fire as at user select)
- */
+* @typedef { CustomStylesObject } CustomStylesObject
+ * @property {React.CSSProperties | string} react-dadata__custom-action - [optional] classname or CSSProperties for Custom Action component (in Suggestions List)
+ * @property {React.CSSProperties | string} react-dadata__suggestion - [optional] CSS classname or CSSProperties for Suggestion Item component (in Suggestions List)
+ * @property {React.CSSProperties | string} react-dadata__suggestion-note - [optional] CSS classname or CSSProperties for Note component (of Suggestion Item)
+ * @property {React.CSSProperties | string} react-dadata__suggestions - [optional] CSS classname or CSSProperties for Suggestions List component
+**/
 
 /**
  * @typedef { BaseProps } basic collect of props with predefined types independent of query mode type
+ * @property  { boolean } allowClear - [optional] show/hide clear fieldd control
+ * @property  { boolean } autocomplete - [optional] property translated to native input tag;
+ * @property  { boolean } city - [optional] optional to city-mode
+ * @property  { string } className - [optional] additional classname
+ * @property  { CustomStylesObject } customStyles - [optional] customize styling of individual specific components (parts)
+ * @property  { number } count - [optional] single query limit (default: 10)
+ * @property  { boolean | object } customEndpoint - [optional] optional uri to fetch suggestion's (to proxy scenario or local hosted DaData service), may be string: full or relative uri or object with 'host' and/or 'api' property
+ * @property  { customInput } customInput - [optional] function that fires with one argument { BaseInputProps }
+ * @property  { debounce } debounce - [optional] debounce to onChange handler (default: 350 ms).
+ * @property  { boolean } forceOpenList - [optional] prevent to close opened suggestions list (by Blur event or user select value Action); it also is way may to debug suggestions
+ * @property  { onIdleOut } onIdleOut - [optional] onIdleOut handler, fires with one argument - current query, when by this query has not returned suggestions
+ * @property  { object | function } payloadModifier - [optional] object to patch payload or function returned payload object to send, that has auto generated payload object as argument
+ * @property  { string } placeholder - [optional] placeholder
+ * @property  { string } query - [optional] query for search
+ * @property  { boolean } showNote - [optional] show/hide note at suggestions list
+ * @property  { boolean } silentQuery - [optional] initial query that not showed on input but determine  suggestion list that showed at first
+ * @property  { React.CSSProperties } style - [optional] custom styling
+ * @property  { string } token - [required] API authorization token
+ * @property  { FetchType } type - [optional] specifics fetching by data type groups
+ *
+ * @property  { customActions } customActions - [optional] adding custom action to base suggestions dropdown list
+ * @property  { onChange } onChange - [optional] onChange handler
+ * @property  { function } silentInit - [optional] function that may be used to autoselect from preventive fetched (by placed query or silentQuery), it called with list of fetched suggestions, and if it will return index, appropriate suggestion will be selected (all handlers fire as at user select)
  */
+
 type BaseProps = {
   allowClear?: boolean;
   autocomplete?: 'on' | 'off';

--- a/index.d.ts
+++ b/index.d.ts
@@ -1,6 +1,6 @@
 // tslint:disable:max-line-length
 import * as React from 'react';
-import { ReactElement } from 'react';
+import {ReactElement, SyntheticEvent} from 'react';
 
 // types formed by description's from:
 // - https://dadata.ru/api/find-address/#response
@@ -456,6 +456,7 @@ type BaseProps = {
   customInput?: (props: BaseInputProps) => React.ReactNode;
   debounce?: number;
   forceOpenList?: boolean;
+  onBlur?: (event: SyntheticEvent<HTMLInputElement, FocusEvent>, query: string) => void;
   onIdleOut?: (query: string) => void;
   payloadModifier?: object | ((payload: BasePayload) => BasePayload & object);
   placeholder?: string;

--- a/package.json
+++ b/package.json
@@ -26,7 +26,7 @@
     "react-dom": "^17.0.2"
   },
   "dependencies": {
-    "react-highlight-words": "^0.12.0"
+    "react-highlight-words": "^0.18.0"
   },
   "devDependencies": {
     "@types/react": "^18.0.17",

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "react-dadata-box",
-  "version": "1.3.7",
+  "version": "1.3.8",
   "description": "Компонент для работы с DaData",
   "main": "dist/index.js",
   "style": "dist/index.css",

--- a/src/index.js
+++ b/src/index.js
@@ -129,25 +129,6 @@ const renderCustomActions = ({ customActions, customStyles, suggestions }, muteE
 
   let actions = customActions instanceof Function ? customActions(suggestions) : customActions;
 
-  // ToDo: @remove in >= 1.3.5
-  if (!(customActions instanceof Function)) {
-    console.warn(
-      '\x1b[31m' +
-        `
-   ╭────────────────────────────────────────────────────────────────╮
-   │                  react-dadata-box@1.3.4                        │
-   │                *** DEPRECATION WARNING ****                    │
-   │  at v1.3.5 will be deprecated variant to place customActions   │
-   │   as React.Element it must be placed only as function that     │
-   │    returns React.Element and take suggestions as argument      │
-   │               see more in project README.md                    │
-   │      https://github.com/orbita-center/react-dadata-box         │
-   ╰────────────────────────────────────────────────────────────────╯
-    ` +
-        '\x1b[0m'
-    );
-  }
-
   actions = actions instanceof Array ? actions : actions ? [actions] : false;
 
   return actions && actions.length

--- a/src/index.js
+++ b/src/index.js
@@ -242,8 +242,13 @@ class ReactDaDataBox extends React.PureComponent {
     }
   };
 
-  onInputBlur = () => {
+  onInputBlur = (e) => {
     this.setState({ inputFocused: false });
+    if (e && this.props.onBlur) {
+      e.preventDefault();
+      e.stopPropagation();
+      this.props.onBlur(e, this.state.query);
+    }
   };
 
   debounce = (func, cooldown = 350) => {
@@ -459,6 +464,7 @@ ReactDaDataBox.propTypes = {
   debounce: PropTypes.number,
   forceOpenList: PropTypes.bool,
   onChange: PropTypes.func,
+  onBlur: PropTypes.func,
   onIdleOut: PropTypes.func,
   payloadModifier: PropTypes.oneOfType([PropTypes.object, PropTypes.shape, PropTypes.func]),
   placeholder: PropTypes.string,

